### ISSUE TITLE
BUG - sev-2 - iOS (and maybe Android) - Secure messaging: user report of unable to edit text with VoiceOver

### ISSUE THREAD

**Issue by TKDickson** [department-of-veterans-affairs/va-mobile-app#8042]
## What happened?

[Veteran report from DSVA slack](https://dsva.slack.com/archives/C018V2JCWRJ/p1708554356928879):

> Hello mobile team. I have a bug report from our friend Claudia, the blinded Veteran who we meet with monthly. The issue is with the text field when she composes a new Secure Message 
> 
> To help summarize this a bit: Claudia was using Messaging in the app on iOS with VoiceOver. When they dictate a message, they’re unable to get back into the field and edit the message by moving around. They can go word by word or select all the text and remove it, but they don’t want to do that. They said they’re able to do this in another app, for example, Outlook email. It’s possible this is an Apple bug, as Claudia mentioned her phone is newer. 

## Specs:

- Device:
- OS:
- User Account:
- Accessibility State:

## Steps to Reproduce

**Transcript from the meeting with the veteran:**
My transcript from the meeting we just had. She is very frustrated, this was working fine in previous releases.
Claudia – Frustrating problems with secure messaging and editing the message
Is it the iPhone 15 Pro or what
not editing – cannot edit the message
sending it all incorrect
dictating makes mistakes
I look like an idiot bc I cannot fix the errors
Lets me go word by word but I cannot edit
It used to be fine
It makes a weird thunk sound when you’re opening it up
I have the latest update
I hate 15 Pro
The Voice Over is crappy now
The text message only gives you partial then gives silent
Gotten to I don’t care – that’s an apple issue
I’m not typing in other apps – they are just buttons
In this one I’m typing a lot
Outlook app works fine – lets me edit
I can only delete one character at the end
I can only select all text – I can’t undo the select
It was fine before
It was an update or what happened
Don’t know if it’s the upgrade
This is the native app
I’m a champion!

## Bug Severity - BE SURE TO ADD THE SEVERITY LABEL
 
See [Bug Tracking](https://department-of-veterans-affairs.github.io/va-mobile-app/docs/QA#issue-severity) for details on severity levels

- Impact: High Low 
- Frequency: High Low 

- 3 - Low
- 2 - High
- 1 - Critical

## Screen shot(s) and additional information

<details>
  <summary>Full JSON response for services related to issue (expand/collapse)</summary>
  

</details>

## Ticket Checklist
- [ ] Steps to reproduce are defined
- [ ] Desired behavior is added
- [x] Labels added (front-end, back-end, relevant feature, accessibility, etc)
- [x] Estimate of 1 added (for front-end tickets)
- [x] Added either to the relevant feature epic (if found during new feature implementation), or the [bug epic](https://app.zenhub.com/workspaces/va-mobile-product-view-610035bc5395bb000e62e529/issues/gh/department-of-veterans-affairs/va-mobile-app/5367)

**Comment by TKDickson** [comment]
We have previously gotten reports of this before, and fixed it: #2422

**Comment by rbontrager** [comment]
Unable to reproduce this on an iPhone 13 running 17.3.1 with the RC candidate build for 2.21.  I'm able to go word for word and edit as I'm going along ([8042](https://drive.google.com/file/d/1ydVP1voxoABby2jx1c1lCMF44wmgbx6P/view?usp=sharing)). Still waiting for the latest dev build to work to see if I can reproduce it there.

**Comment by DJUltraTom** [comment]
@rbontrager @TKDickson  should we close this as a "cannot reproduce"?

**Comment by Sparowhawk** [comment]
@DJUltraTom @rbontrager @TKDickson are we closing this as cannot reproduce?

**Comment by brea11y** [comment]
Just an update that we spoke to Claudia again today during the BVA meeting and she is still experiencing this issue on her iPhone 15 Pro. Another user in that meeting also said that they have the same experience on iPhone, but did not mention what iPhone model they are using.

**Comment by bischoffa** [comment]
@ala-yna @rbontrager this Sev-2 has been around for 2 months and the issue still persist.  No update since Feb.

**Comment by brea11y** [comment]
@bischoffa – The only update we've received is from the user herself (and another user in the BVA meeting) last week that said the problem is still happening. After that meeting, I went into the app and tried to recreate it. The first time I went through the steps that Claudia mentioned, I experienced issues with being able to edit also. The second and third time I went through, I had zero issue and it seemed to work fine. It just seemed very finicky and not as smooth as other apps that Claudia also mentioned working fine for her (Gmail, Outlook, etc.). I also tried iMessage, Apple Mail, and the Healow app (a healthcare app that offers secure messaging) and had no issues editing in those. It definitely seems to be something in our app that wasn't happening prior to a few months ago, but it also doesn't seem to be happening every single time I try it.

**Comment by Sparowhawk** [comment]
@brea11y @bischoffa @rbontrager @TKDickson This may have been introduced in September

https://www.applevis.com/blog/ios-17-ipados-17-accessibility-report-voiceover-braille-low-vision-issues-improvements

Looks like Apple is aware and is working on an active ticket for it, but their suggested workaround is to go to the users phone settings and change the gesture timing settings.

The only other thing noted by Apple is that it might be a 3rd party keyboard issue and their recommendation is to uninstall and use the native iOS keyboard and voice over controls ifi they aren't already.

**Comment by Sparowhawk** [comment]
Unfortunately there doesn't appear to be anything that we can do for this bug.

**Comment by brea11y** [comment]
Thanks so much for looking into this further, @Sparowhawk. I'll forward this on to Martha so that we can let Claudia know. Hopefully Apple fixes it on their end soon. 🤞🏼

**Comment by bischoffa** [comment]
@Sparowhawk thoughts on moving this to blocked and revisiting this when apple fix is implemented? Or close it and follow-up later (just don't want it to get lost). @rbontrager how have we handled tickets like this in the past

**Comment by bischoffa** [comment]
@Sparowhawk @rbontrager checking in from last week's comment.

**Comment by rbontrager** [comment]
@bischoffa I feel like we usually move it to blocked but I don't think I've had a ticket like this before.

**Comment by bischoffa** [comment]
@rbontrager go ahead and move it to blocked and put blocked external label on it.

**Comment by bischoffa** [comment]
@Sparowhawk has Apple implemented their fix yet? If not do we have any ETA?

**Comment by bischoffa** [comment]
@Sparowhawk has Apple implemented their fix or provide any kind of update? CC @ala-yna

**Comment by ala-yna** [comment]
Discussed this at refinement at 8/27/24. We're hoping this gets fixed in iOS 18. Will check back in on this finding once the new iOS comes out.

**Comment by timwright12** [comment]
> We're hoping this gets fixed in iOS 18. Will check back in on this finding once the new iOS comes out. Projected release date is September 18th.

@ala-yna any update on this?

**Comment by Sparowhawk** [comment]
Apple is claiming this is fixed from what I can tell as I don't see any bug reports for iOS 18 for this issue. I did find a potential work around for vets if this is a continued issue though.

It does involve them changing their keyboard from English(US) to a different English keyboard in their settings.

**Comment by DJUltraTom** [comment]
unable to reproduce using iphone XS ios 18.0.1
able to enter text, leave field, comback and edit text, using the rotor I can select to edit and navigate the previously entered text by words or characters.
Closing as "fixed" since it seems apple resolved this
